### ISSUE TITLE
Feature request: Other input formats

### ISSUE THREAD

**Issue by CMDR-Cometborne** [CMDR-Kiel42/EDMC_SpanshRouter#6]
Would it be possible to support other input formats than the .csv exported by spansh, as this would make the plug-in usable for a lot of different tasks?

The first and most obvious format would be a text file that simply contains many system names, one name per line.  It would no longer be possible to display a jump count estimate, but this allows using the plugin for any kind of routing:
- visiting all systems of a faction for BGS work
- buckyball racing
- writing on galmap

The second format is a variant of this where each line contains a system name, followed by a slash and a station name.  In this case, the plugin should put just the system name on the clipboard, and display system and station name in the window.  This format helps with visiting the stations with the oldest market data, or hunting for CMM and other rare commodities for Operation IDA.

The third format is the output of the EDTS route optimizer.  EDTS is a set of python scripts by CMDR Alot that contains a travelling salesman optimizer.  You give it a start and end system and a jump range, and it will produce an optimized route for visiting all the systems.  I have attached a sample output.  The first system in the list is on a line by its own, then you have a long list of lines containing further way points with distance and jump count estimate, then an empty line, and two lines with jump and SC totals.

Each way point name is either 
- just a system name, 
- a combination of <system name>/<station name>
- several of the above separated by commas

For the jump count estimate, I recommend taking the higher number of EDTS's estimates; if there are several names in one line, estimate one jump for each extra system.

EDTS output: [sample.txt](https://github.com/CMDR-Kiel42/EDMC_SpanshRouter/files/3414464/sample.txt)

Oh, and it would be really good if you could ensure that your plugin can deal with CRLF (Windows), LF (Unix), and CR (Mac) end-of-line markers.

**Comment by CMDR-Kiel42** [comment]
Yup, all of that is totally doable. Those are great ideas, I'll start working on it.
About the end-of-line markers, are you asking because you ran into some issues? If you did, could you send me the file that caused it?

**Comment by CMDR-Cometborne** [comment]
Thanks for the positive feedback on my ideas.  The remark about the end-of-line markers was not triggered by a malfunction, this is more a general concern (or even pet peeve) I have because I do most of my planning on a FreeBSD box, which generates files with Unix-style EOL.

**Comment by CMDR-Kiel42** [comment]
@CMDR-Cometborne If you want to beta-test the EDST format compatibility, feel free to download [this version](https://github.com/CMDR-Kiel42/EDMC_SpanshRouter/tree/other_formats). It's the branch in which I'm currently working, the EDST and simple text formats should already work. I'll add the text format with station probably later this week.

**Comment by vStone** [comment]
It might be overkill to parse the EDST format since edst.py also has a `--format csv` option and output to json using `-J`.

**Comment by CMDR-Cometborne** [comment]
After some local hardware issues, I was finally able to test the new branch, and it is working just fine.  Thumbs up!

**Comment by CMDR-Cometborne** [comment]
Not sure if it is the update to Windows 10 19H03 or EDMC 3.43, but the format selection is no longer readable.  After clicking "import file", I see this:
![image](https://user-images.githubusercontent.com/25147462/66023726-4b6a7f80-e4f2-11e9-80af-76854c9a64fe.png)

If I click and hold over one of the entries, I can see the text is there, but it is black text on black background unless I select:
![image](https://user-images.githubusercontent.com/25147462/66023843-a56b4500-e4f2-11e9-9f1a-800e5386fa61.png)

**Comment by CMDR-Kiel42** [comment]
Sorry for the lack of updates lately, real life caught up with me!
I kinda agree with @vStone, but since the code is already here and seems to be working, might as well keep it. I'll try to release the update this weekend
I'll check this last issue as soon as I can.
